packages/mask/src/plugins/Trader/SNSAdaptor/trader/Trader.tsx

```diff
@@ -114,12 +111,28 @@ export function Trader(props: TraderProps) {
             dispatchTradeStore({
                 type: AllProviderTradeActionType.UPDATE_OUTPUT_TOKEN,
                 token: coin.contract_address
-                    ? createERC20Token(chainId, coin.contract_address, decimals, coin.name, coin.symbol)
+                    ? createERC20Token(chainId, coin.contract_address, coin.decimals, coin.name, coin.symbol)
                     : undefined,
             })
         }
     }, [coin, NATIVE_TOKEN_ADDRESS, inputToken, outputToken, currentChainId, targetChainId, decimals])
 
+    useEffect(() => {
+        if (!defaultInputCoin) return
+        dispatchTradeStore({
+            type: AllProviderTradeActionType.UPDATE_INPUT_TOKEN,
+            token: defaultInputCoin.contract_address
+                ? createERC20Token(
+                      chainId,
+                      defaultInputCoin.contract_address,
+                      defaultInputCoin.decimals,
+                      defaultInputCoin.name,
+                      defaultInputCoin.symbol,
+                  )
+                : undefined,
+        })
+    }, [defaultInputCoin])
```

Comment:
does it depend on `chainId`?

Reply:
I think so, chainId was used.

Reply:
Just for business. Adding the chainId will cause a bug in the dependency